SpiceSharp/Components/Subcircuits/Behaviors/Biasing/InitialConditionBehavior.cs

```diff
@@ -0,0 +1,66 @@
+﻿using SpiceSharp.Behaviors;
+using SpiceSharp.Entities;
+using System.Threading.Tasks;
+
+namespace SpiceSharp.Components.SubcircuitBehaviors
+{
+    /// <summary>
+    /// Initial condition behavior for a <see cref="Subcircuit"/>.
+    /// </summary>
+    /// <seealso cref="SubcircuitBehavior{T}" />
+    /// <seealso cref="IInitialConditionBehavior" />
+    public class InitialConditionBehavior : SubcircuitBehavior<IInitialConditionBehavior>, IInitialConditionBehavior
+    {
+        private InitialConditionParameters _icp;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InitialConditionBehavior"/> class.
+        /// </summary>
+        /// <param name="name">The identifier of the behavior.</param>
+        /// <remarks>
+        /// The identifier of the behavior should be the same as that of the entity creating it.
+        /// </remarks>
+        public InitialConditionBehavior(string name) : base(name)
+        {
+        }
+
+        /// <summary>
+        /// Bind the behavior to a simulation.
+        /// </summary>
+        /// <param name="context">The binding context.</param>
+        public override void Bind(BindingContext context)
+        {
+            base.Bind(context);
+            context.Behaviors.Parameters.TryGetValue(out _icp);
+        }
+
+        /// <summary>
+        /// Sets the initial conditions for the behavior.
+        /// </summary>
+        public void SetInitialCondition()
+        {
+            if (_icp != null && _icp.ParallelInitialCondition && Behaviors.Length > 1)
+            {
+                var tasks = new Task[Behaviors.Length];
+                for (var t = 0; t < Behaviors.Length; t++)
+                {
+                    var task = t;
+                    tasks[t] = Task.Run(() =>
```

Comment:
@svenboulanger Consider introducing some abstraction for running tasks in parallel and use it everywhere. So for example in the near feature could think about distributed computation of tasks.

Reply:
@svenboulanger Also some abstraction for executing tasks in a sequence (without parallel)

Reply:
I'll have another look at it. 👍